src/Container.php

```diff
@@ -78,6 +87,23 @@ public function add($alias, $concrete = null, $singleton = false)
     /**
      * {@inheritdoc}
      */
+    public function addServiceProvider($provider)
+    {
+        if ((! is_string($provider)) && (! $provider instanceof ServiceProvider)) {
+            throw new \InvalidArgumentException(
+                'When registering a service provider, you must provide either and instance of ' .
+                '[\League\Container\ServiceProvider] or a fully qualified class name'
+            );
+        }
+
+        $this->providers[] = $provider;
+
+        return $this;
```

Comment:
Was mainly thinking of chaining as you'd most likely be registering multiple service providers